code/modules/power/singularity/singularity.dm

```diff
@@ -285,32 +285,44 @@
 	src.energy += A.singularity_act(src, current_size)
 	return
 
-/obj/singularity/proc/move(var/force_move = 0)
+/obj/singularity/proc/move(force_move_direction = 0)
 	if(!move_self)
-		return 0
+		return FALSE
 
 	var/movement_dir = pick(GLOB.alldirs - last_failed_movement)
 
-	if(force_move)
-		movement_dir = force_move
+	if(force_move_direction)
+		movement_dir = force_move_direction
 
 	if(target && prob(60))
 		movement_dir = get_dir(src,target) //moves to a singulo beacon, if there is one
+	else
+		var/location
+		if(prob(16))
+			location = GetAbove(src)
+			if (location)
+				src.Move(location, UP)
+				return TRUE
+		else if(prob(16))
```

Comment:
If I still remember how to calculate probabilities, the probability that the singulo will move down is only 0.84 * 0.16=0,1344% (i.e. the chance that it moves up takes away from the chance that it will move down).
For a fair chance in either direction you should do (semi-pseudo-code):
``` 
if(prob(16))
    if(prob(50) && GetAbove(src))
        MoveUp()
    else if(GetBelow(src)
        MoveDown()
return TRUE
```